### DISCUSSION TITLE
How can i detect which block or entity a player is looking at?

### DISCUSSION THREAD

**Discussion by pewy007** [Minestom/Minestom#1084] [Help]
In Paper I use Raytracing, what about MInestom? *Sorry, for my bad English*

**Comment by KrystilizeNevaDies** [comment]
`LivingEntity#getTargetBlockPosition` Will give you the block the player is looking at.
However for the entity that a player is looking at, you will need an external library. e.g. https://github.com/EmortalMC/Rayfast
